CNKI.js

```diff
@@ -39,32 +39,32 @@
 // fetches Refworks record for provided IDs and calls next with resulting text
 // ids should be in the form [{dbname: "CDFDLAST2013", filename: "1013102302.nh"}]
 function getRefworksByID(ids, next) {
-	var postData = "";
-	for (var i=0, n=ids.length; i<n; i++) {
+	for(var i = 0; i < ids.length; i++) {
+		var postData = "";
 		postData += ids[i].dbname + "!" + ids[i].filename + "!1!0,";
+		postData = "formfilenames=" + encodeURIComponent(postData);
+		postData += '&hid_kLogin_headerUrl=/KLogin/Request/GetKHeader.ashx%3Fcallback%3D%3F';
+		postData += '&hid_KLogin_FooterUrl=/KLogin/Request/GetKHeader.ashx%3Fcallback%3D%3F';
+		postData += '&CookieName=FileNameS';
+		
+		ZU.doPost('https://kns.cnki.net/kns/ViewPage/viewsave.aspx?displayMode=Refworks', postData, 
+			function(text) {
+				var parser = new DOMParser();
+				var html = parser.parseFromString(text, "text/html")
+				var text = ZU.xpath(html, "//table[@class='mainTable']//td")[0].innerHTML;
+				var text = text.replace(/<br>/g, '\n');
+				text = text.replace(/^RT\s+Dissertation\/Thesis/gmi, 'RT Dissertation');
+				text = text.replace(/^(A[1-4]|U2)\s*([^\r\n]+)/gm, 
+						function(m, tag, authors) {
+							var authors = authors.split(/\s*[;，,]\s*/); //that's a special comma
+							if (!authors[authors.length-1].trim()) authors.pop();
+							return tag + ' ' + authors.join('\n' + tag + ' ');
+						}
+					);
+				next(text);
+			}
+		);
```

Comment:
What was the reason for changing this? Calling 'doPost" within a loop isn't ideal, because it results in all requests being made at the same time (or possibly limited by a per-server connection limit in the network layer) rather than run serially. While this can be faster, it makes sites more likely to block requests.

Reply:
CNKI post API has changed in the search results. A new parameter is added, it looks like this parameter is from cookies. There is less information for me to figure out how this parameter is generated.

Reply:
I just mean the change in the last commit from a single POST request to multiple requests. Is it no longer possible to make a single POST request? If not, it'd still be better to run this serially by `shift()`ing off a list of ids and processing each one in turn until there are no more, at which point `next` can be called. I can provide an example if necessary. (We're working on `async`/`await` support in translators to make this easier, but for now it has to be done the awkward callback way.)

Reply:
Let me know if you want help with this. I'd love to get this fix in, but I don't want to release a version that will make up to 50 simultaneous POST requests and potentially get people blocked by the site.

Reply:
@dstillman Sorry for the delay, I was in the hospital a few days ago. An encrypted parameter is needed when getting multiple items by a single POST. I don't know how the string is encrypted, maybe this encrypted string is generated based on cookies on search page.  I hope you can give me some examples of `async`/`await` in translators.

Reply:
Oh, sorry to hear that.

I'm not suggesting making a single POST. I'm just saying this needs to be structured so that the individual POST requests are serialized instead of run concurrently, in the way I describe above. Let me know if you need an example of that.

`async`/`await` isn't yet available.

Reply:
Actually, I'll take care of this. I'll have an update soon.

Reply:
@dstillman Sorry, I don't quite understand what you mean.  I am a beginner in javascript 😂.  I don't know how to send posts in serialized. I hope you can give me a hand 😊

Reply:
I already fixed this and committed the translator. See my message below.